### PR TITLE
kops: Add kops AWS PR builder

### PR DESCRIPTION
**Pull request by zmerlynn**
This is the PR builder for the Kubernetes repo for AWS-via-kops. After
I get this working, I'll implement the PR builder for the kops repo as
well (which can use this same script, but I need to fix something in
the e2e image first).

Along the way: bump `kops-node` to 4 in the main test run as well.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/979)

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @kubernetes/sig-aws

**Comment by kris-nova** [comment]
Woo! This looks great!

File: jobs/pull-kubernetes-e2e-kops-aws.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+readonly version=$(source build-tools/util.sh && echo $(kube::release::semantic_version))
+export KUBERNETES_PROVIDER="kops-aws"
+
+export KOPS_STATE_STORE="${KOPS_STATE_STORE:-s3://k8s-kops-jenkins/}"
+export KOPS_CLUSTER_DOMAIN="${KOPS_CLUSTER_DOMAIN:-test-aws.k8s.io}"
+export E2E_NAME="aws-kops-${NODE_NAME}-${EXECUTOR_NUMBER:-0}"
+export E2E_OPT="${E2E_OPT:-}\
+  --kops-cluster ${E2E_NAME}.${KOPS_CLUSTER_DOMAIN}\
+  --kops-kubernetes-version https://storage.googleapis.com/${KUBE_GCS_RELEASE_BUCKET}/ci${KUBE_GCS_RELEASE_SUFFIX}/${version}\
+  --kops-nodes 4\
```
Comment (spxtr) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: It looks like this is duplicated?

**Comment by spxtr** [comment]
/lgtm

Before setting `always_run` to true you should verify that it is stable and runs fairly quickly.

File: jobs/pull-kubernetes-e2e-kops-aws.sh (hunk 1 above)
Comment (zmerlynn) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: Where did you see the other? I bumped https://github.com/kubernetes/test-infra/pull/979/files#diff-cc56a431187d0d372655f10b42900f84, but that's the normal job.

**Pull request merged**